mirascope/integrations/_middleware_factory.py

```diff
@@ -51,6 +51,12 @@ def middleware_factory(
         [SyncFunc | AsyncFunc], AbstractContextManager[_T]
     ] = default_context_manager,
     custom_decorator: Callable | None = None,
+    handle_error: Callable[[Exception, SyncFunc | AsyncFunc, _T | None], Any]
```

Comment:
note to self: this is a breaking change since we don't currently require named arguments, so this has to be moved to the end of the definition